### PR TITLE
PublicDashboards: collect stats for public dashboards

### PR DESCRIPTION
**Pull request by evictorero**
**What this PR does / why we need it**:
Collect and send internal stats for Public dashboards.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise-partnerships-team/issues/205

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/sqlstore/stats.go [hunk 1]
```diff
@@ -105,6 +105,9 @@ func (ss *SQLStore) GetSystemStats(ctx context.Context, query *models.GetSystemS
 		sb.Write(`(SELECT COUNT(*) FROM ` + dialect.Quote("data_keys") + `) AS data_keys,`)
 		sb.Write(`(SELECT COUNT(*) FROM ` + dialect.Quote("data_keys") + `WHERE active = true) AS active_data_keys,`)
 
+		// TODO: table name will change and filter should check only for is_enabled = true
+		sb.Write(`(SELECT COUNT(*) FROM ` + dialect.Quote("dashboard_public_config") + `WHERE 1 = 1) AS public_dashboards,`)
```
Comment (owensmallwood) [inline pkg/services/sqlstore/stats.go:109]: Whats the point of `WHERE 1=1`

File: pkg/infra/metrics/metrics.go [hunk 2]
```diff
@@ -550,6 +553,12 @@ func init() {
 		Help:      "total amount of data keys in the database",
 		Namespace: ExporterName,
 	}, []string{"active"})
+
+	MStatTotalPublicDashboards = prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (owensmallwood) [inline pkg/infra/metrics/metrics.go:557]: So a prometheus "Gauge" is a type of metric that can be exposed?

**Review by owensmallwood — APPROVED** [review]
Looks good! Thanks Eze 🥳

File: pkg/services/sqlstore/stats.go (hunk 1 above)
Comment (evictorero) [reply pkg/services/sqlstore/stats.go:109]: As a reminder, this PR is not ready to merge, I need to make changes in this line and remove the comment above after the final DB schema definition.

**Pull request merged**